src/components/protocol_handler/test/multiframe_builder_test.cc

```diff
@@ -417,71 +417,58 @@ TEST_F(MultiFrameBuilderTest, Add_ConsecutiveFrames_OneByOne) {
   }
 }
 
-// TODO(OHerasym) : Test do not finish working
 TEST_F(MultiFrameBuilderTest, Add_ConsecutiveFrames_per1) {
   AddConnections();
   ASSERT_FALSE(test_data_map_.empty());
   // After processing each frame we remove it from messageId_it
   // After processing all session data - it removes from session_map
   // and so on
-  // TODO(Ezamakhov): optimize speed of test by skipping erasing data
-  while (!test_data_map_.empty()) {
-    MultiFrameTestMap::iterator connection_it = test_data_map_.begin();
-    while (connection_it != test_data_map_.end()) {
-      SessionToMutiframeDataTestMap& session_map = connection_it->second;
-
-      SessionToMutiframeDataTestMap::iterator session_it = session_map.begin();
-      while (session_it != session_map.end()) {
-        MessageIDToMutiframeDataTestMap& messageId_map = session_it->second;
-
-        MessageIDToMutiframeDataTestMap::iterator messageId_it =
-            messageId_map.begin();
-        while (messageId_it != messageId_map.end()) {
-          MutiframeData& multiframe_data = messageId_it->second;
-          ProtocolFramePtrList& multiframes = multiframe_data.multiframes;
-          ASSERT_FALSE(multiframes.empty());
-
-          const ProtocolFramePtr frame = multiframes.front();
-          ASSERT_TRUE(frame);
-
-          EXPECT_EQ(RESULT_OK, multiframe_builder_.AddFrame(frame))
-              << "Frame: " << frame;
+  MultiFrameTestMap::iterator connection_it = test_data_map_.begin();
+  while (test_data_map_.end() != connection_it) {
+    SessionToMutiframeDataTestMap& session_map = connection_it->second;
```

Comment:
@VProdanov It is possible but it will make the code more complicated. There is three level nested loop which means three `for_each`, three functors and code spread between them. This will be more difficult to read than current loops.